lib/auth/plain.js

```diff
@@ -1,176 +1,42 @@
 'use strict';
 
-var f = require('util').format,
-  retrieveBSON = require('../connection/utils').retrieveBSON,
-  Query = require('../connection/commands').Query,
-  MongoError = require('../error').MongoError;
+const retrieveBSON = require('../connection/utils').retrieveBSON;
+const Query = require('../connection/commands').Query;
+const AuthProvider = require('./authProvider').AuthProvider;
 
-var BSON = retrieveBSON(),
-  Binary = BSON.Binary;
-
-var AuthSession = function(db, username, password) {
-  this.db = db;
-  this.username = username;
-  this.password = password;
-};
-
-AuthSession.prototype.equal = function(session) {
-  return (
-    session.db === this.db &&
-    session.username === this.username &&
-    session.password === this.password
-  );
-};
+// TODO: can we get the Binary type from this.bson instead?
```

Comment:
agree with comment, I think we should actually only ever use `retrieveBSON` in one place, and then use the bson passed down the chain for everything else

Reply:
This may have to wait until we upgrade mongodb-core to the latest version of BSON. currently `this.bson` is an instance of `BSON`, so it does not have the static members that are defined on `BSON`. Unless we want to do `this.bson.constructor.Binary`?